### PR TITLE
Support to build cpp client lib

### LINKED ISSUE #956: Support to use cpp client
Now the `cpp client` can't use it separate nebula source code, we should package it to an rpm/deb package

Step1 :  we should wrap a layer out of `cpp client` , make it to a .so file
Step2:  package it to rpm/deb

### PR DESCRIPTION
**Pull request by laura-ding**
- build cpp client to so file
- add UTs
- add how to use the interface example
- package cpp client so to a deb/rpm file

close #956

### CODE REVIEW

**Comment by laura-ding** [comment]
jenkins go

File: src/client/cpp/include/nebula/NebulaClient.h [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+class NebulaClientImpl;
+
+class NebulaClient {
+public:
+    NebulaClient(const std::string& addr, uint16_t port);
+    virtual ~NebulaClient();
+
+    // must be call on the front of the main()
+    static void init(int argc, char *argv[]);
```
Comment (dutor) [inline src/client/cpp/include/nebula/NebulaClient.h]: Please consider that the end user might have multiple instances of NebulaClient. Try not to use global parameters, such as `timeout`.

File: src/client/cpp/lib/NebulaClientImpl.cpp [hunk 2]
```diff
@@ -0,0 +1,183 @@
+/* Copyright (c) 2019 vesoft inc. All rights reserved.
+ *
+ * This source code is licensed under Apache 2.0 License,
+ * attached with Common Clause Condition 1.0, found in the LICENSES directory.
+ */
+
+#include "NebulaClientImpl.h"
+#include "base/Base.h"
```
Comment (dutor) [inline src/client/cpp/lib/NebulaClientImpl.cpp]: Place `Base.h` at the first line.

File: src/client/cpp/lib/NebulaClientImpl.h [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+#include "gen-cpp2/GraphServiceAsyncClient.h"
+#include <string>
+
+namespace nebula {
+namespace graph {
+
+class NebulaClientImpl {
+public:
+    NebulaClientImpl(const std::string& addr, uint16_t port);
+    virtual ~NebulaClientImpl();
```
Comment (dutor) [inline src/client/cpp/lib/NebulaClientImpl.h]: ditto.

**Comment by laura-ding** [comment]
jenkins go

**Comment by laura-ding** [comment]
jenkins go

File: src/client/cpp/example/README.md [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+```
+
+package **DEB**
+
+```bash
+bash> cd package
+bash> ./package_libcppclient.sh ${VERSION} DEB
+```
+
+after finish package, install the rpm or deb file
```
Comment (Amber1990Zhang) [inline src/client/cpp/example/README.md]: After

File: src/client/cpp/example/README.md [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+
+## How to use in your code
+
+- Step 1: init arvs(a process can only call once)
+- Step 2: init connection pool, the default connections is 10, timeout is 1000 ms (a process can only call once)
```
Comment (Amber1990Zhang) [inline src/client/cpp/example/README.md]: the default connection number

**Comment by dangleptr** [comment]
Do we want the cpp graph client  in 1.x.    
@sherman-the-tank @whitewum

**Comment by dangleptr** [comment]
After talk with @sherman-the-tank offline,  we could take it in 1.x.  
Do you mind to rebase it on current master? @laura-ding Thanks.

**Comment by laura-ding** [comment]
> After talk with @sherman-the-tank offline, we could take it in 1.x.
Do you mind to rebase it on current master? @laura-ding Thanks.

@dangleptr ok, I will rebase it soon.

**Comment by doublex** [comment]
Great news!

File: src/client/cpp/lib/CMakeLists.txt [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+    $<TARGET_OBJECTS:network_obj>
+    $<TARGET_OBJECTS:meta_client>
+    $<TARGET_OBJECTS:time_obj>
+    $<TARGET_OBJECTS:stats_obj>
+    $<TARGET_OBJECTS:gflags_man_obj>
+    $<TARGET_OBJECTS:common_thrift_obj>
+    $<TARGET_OBJECTS:thrift_obj>
+    $<TARGET_OBJECTS:graph_thrift_obj>
+    $<TARGET_OBJECTS:meta_thrift_obj>
+    $<TARGET_OBJECTS:storage_thrift_obj>
```
Comment (Shylock-Hg) [inline src/client/cpp/lib/CMakeLists.txt:32]: There are maybe some unused object and libraries.

**Comment by CPWstatic** [comment]
We had a cpp client in v2.0.

**Pull request closed without merging**